sdk/src/common/env_variables.cc

```diff
@@ -0,0 +1,157 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+#include "opentelemetry/sdk/common/env_variables.h"
+
+#ifdef _MSC_VER
+#  include <string.h>
+#  define strcasecmp _stricmp
+#else
+#  include <strings.h>
+#endif
+
+#include <memory>
+
+#include "opentelemetry/sdk/common/global_log_handler.h"
+#include "opentelemetry/version.h"
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace sdk
+{
+namespace common
+{
+
+bool GetRawEnvironmentVariable(const char *env_var_name, std::string &value)
+{
+#if !defined(NO_GETENV)
+  const char *endpoint_from_env = nullptr;
+#  if defined(_MSC_VER)
+  // avoid calling std::getenv which is deprecated in MSVC.
+  size_t required_size = 0;
+  getenv_s(&required_size, nullptr, 0, env_var_name);
+  std::unique_ptr<char> endpoint_buffer;
+  if (required_size > 0)
+  {
+    endpoint_buffer = std::unique_ptr<char>{new char[required_size]};
+    getenv_s(&required_size, endpoint_buffer.get(), required_size, env_var_name);
+    endpoint_from_env = endpoint_buffer.get();
+  }
+#  else
+  endpoint_from_env = std::getenv(env_var_name);
+#  endif  // defined(_MSC_VER)
+
+  if (endpoint_from_env != nullptr)
+  {
+    value = std::string{endpoint_from_env};
+    return true;
+  }
+
+  value = std::string{};
+  return false;
+#else
+  value = std::string{};
+  return false;
+#endif  // !defined(NO_GETENV)
+}
+
+bool GetBoolEnvironmentVariable(const char *env_var_name, bool &value)
+{
+  std::string raw_value;
+  bool exists = GetRawEnvironmentVariable(env_var_name, raw_value);
+  if (!exists)
+  {
+    value = false;
+    return false;
+  }
+
+  if (strcasecmp(raw_value.c_str(), "true") == 0)
```

Comment:
I'm not sure but should it be `_stricmp` in MSVC?

Reply:
> I'm not sure but should it be `_stricmp` in MSVC?

It is, see this code at the start of env_variables.cc:

```
#ifdef _MSC_VER
#  include <string.h>
#  define strcasecmp _stricmp
#else
#  include <strings.h>
#endif
```

Reply:
Thanks, sorry I forgot this.